pfuture.el

```diff
@@ -175,35 +225,58 @@ details see documentation of `accept-process-output'."
      process timeout nil just-this-one))
   (process-get process 'result))
 
+(cl-macrolet
+    ((define-getter (name doc variable )
+       `(define-inline ,name (process)
+          ,doc
+          (declare (side-effect-free t))
+          (inline-letevals (process)
+            (inline-quote
+             (process-get ,',process ',variable))))))
+  (define-getter pfuture-result "Return the output of a pfuture PROCESS." stdout)
+  (define-getter pfuture-stderr "Return the error output of a pfuture PROCESS." stderr))
+
 (defun pfuture-await-to-finish (process)
   "Keep reading the output of PROCESS until it is done.
 Same as `pfuture-await', but will keep reading (and blocking) so long as the
 process is *alive*.
 
 If the process never quits this method will block forever. Use with caution!"
+  ;; If the sentinel hasn't run, disable it. We are going to delete
+  ;; the stderr process here.
+  (set-process-sentinel process nil)
```

Comment:
This breaks `pfuture-await-to-finish` on `pfuture-callback` futures. I'm not sure if that was even supposed to work though?

Reply:
It isn't supposed to work. The callback version is supposed to be a fire-and-forget implementation.